src/NLog.Extensions.Logging/Logging/NLogMessageParameterList.cs

```diff
@@ -79,47 +79,43 @@ private static bool IsValidParameterList(IReadOnlyList<KeyValuePair<string, obje
         {
             hasMessageTemplateCapture = false;
             isMixedPositional = false;
-            isPositional = false;
             originalMessageIndex = null;
-            bool? firstParameterIsPositional = null;
+            isPositional = false;
+            string parameterName;
+
             for (int i = 0; i < parameterList.Count; ++i)
             {
-                if (!TryGetKey(parameterList, ref originalMessageIndex, i, out var parameterKey))
+                if (!TryGetParameterName(parameterList, i, ref originalMessageIndex, out parameterName))
                 {
                     return false;
                 }
 
-                char firstChar = parameterKey[0];
-                if (GetCaptureType(firstChar) != CaptureType.Normal)
+                char firstChar = parameterName[0];
+                if (firstChar >= '0' && firstChar <= '9')
```

Comment:
I think this check is more expensive than before. but maybe not really significant

Reply:
Most users will always use `CaptureType.Normal`. Doing the isPositional-check first makes the code simpler. It will only change performance for those who always use `@` in the parameter-tokens.

Reply:
The goal of this PR is to reduce the complexity of `IsValidParameterList` with minimal impact.